src/jit/lir.cpp

```diff
@@ -1499,13 +1499,19 @@ bool LIR::Range::CheckLIR(Compiler* compiler, bool checkUnusedValues) const
             }
             else if (!def->IsValue())
             {
-                // Stack arguments do not produce a value, but they are considered children of the call.
-                // It may be useful to remove these from being call operands, but that may also impact
-                // other code that relies on being able to reach all the operands from a call node.
-                // The GT_NOP case is because sometimes we eliminate stack argument stores as dead, but
-                // instead of removing them we replace with a NOP.
+// Stack arguments do not produce a value, but they are considered children of the call.
+// It may be useful to remove these from being call operands, but that may also impact
+// other code that relies on being able to reach all the operands from a call node.
+// The GT_NOP case is because sometimes we eliminate stack argument stores as dead, but
+// instead of removing them we replace with a NOP.
+#ifdef _TARGET_ARM_
+                assert(((node->OperGet() == GT_CALL) &&
+                        (def->OperIsStore() || (def->OperGet() == GT_PUTARG_STK) || (def->OperGet() == GT_NOP))) ||
+                       (node->OperGet() == GT_PUTARG_SPLIT && def->OperGet() == GT_PUTARG_STK));
+#else
```

Comment:
This conditions for assertion needs because `node: GT_PUTARG_SPLIT` produce a value, but `def: GT_PUTARG_STK` does not, and `GenTreePutArgSplit` can have `GenTreePutArgStk` as child(operand).